### DISCUSSION TITLE
为网站添加 Google analytics

### DISCUSSION THREAD

**Discussion by excing** [NollGo/Noll#35] [Document]
Noll 是一个针对 GitHub 生态而设计的静态站点生成器，默认支持 `Google Analytics`。用户无需关心 Noll 是如何添加 Google Analytics 的，只需要设置好 `Google Analytics Measurement id`（简称 `GAMID`） 即可直接添加 Google Analytics 功能。

### 设置 GAMID

依托于 GitHub 生态，我们无需编辑脚本文件，只需添加一个环境变量即可让 Google analytics 生效，方法如下：

1. 进入仓库设置页面，点击左侧页面的 `Secrets and variables` 按钮，点击 `Actions` 按钮；
2. 进入 `Actions secrets and variables` 页面，点击 `Variables` Tab；

   ![image](https://user-images.githubusercontent.com/11867809/221804504-60c8f897-1ebe-4f17-ad02-6eeb124ef598.png)

4. 点击页面右上角的 `New repository variable` 按钮，创建一个环境变量，变量名填写为 `GAMID`，变量值为你的 `Google Analytics Measurement id`；

   ![image](https://user-images.githubusercontent.com/11867809/221804309-f56a0dab-9d71-4642-975c-5a6d8c17f687.png)

5. 点击 `New variable` 按钮，完成设置。

通过以上设置，我们的 `NollAction` 会读取 `vars.GAMID` 的值，并把这个值传递给 `Noll`。

### Re-run all jobs

设置完 `varialbe` 后，我们要重新运行一次 `Actions`，方法如下：

进入仓库的 `Actions` 页面，点击最近运行的一次 `Deploy site to Pages` 工作流，如下：

![image](https://user-images.githubusercontent.com/11867809/221806456-7457975e-d5a4-4d33-b581-a68d84a2c12f.png)

进入工作流页面，点击右上角的 `Re-run all jobs` 按钮，点击 `Re-run jobs` 按钮确定，等待重新运行完成。

到此你的站点已经加上了 Google Analytics 功能，尝试访问你的网站，查看 Google Analytics 后台，是不是已经有数据了呢？

**Comment by Agu168** [comment]
按照教程添加 Google Analytics 功能，运行也没有提示问题，但是没有收到数据该怎么办呀？

> 您的网站未启用数据收集功能。如果 48 小时之前您就添加了代码，请确保代码设置正确无误。

**Reply by excing** [reply]
呃，我看了一下你的仓库 `gitblog`，是这个吗？这个好像用的不是 `Noll`，如果你使用的是 `Noll` 生成器，那么你可以等待一会再看，因为 Google Analytics 有一定的延时。

**Reply by Agu168** [reply]
是另外一个名为“blog”的folk，如果是有延时，那就可能是了，早上刚参考教程搞的，后面我再跟踪一下看看，谢谢。
https://github.com/Agu168/blog/discussions/5

**Reply by Agu168** [reply]
这会已经看到Google Analysis的统计结果了，原来是因为延时的原因。
